master/internal/stream/system.go

```diff
@@ -434,14 +502,16 @@ func publishLoop[T stream.Msg](
 
 func doPublishLoop[T stream.Msg](
 	ctx context.Context,
-	pgAddress,
+	dbAddress,
 	channelName string,
 	publisher *stream.Publisher[T],
 ) error {
-	listener, err := newDBListener(pgAddress, channelName)
+	listener, err := newDBListener(dbAddress, channelName)
 	if err != nil {
 		return errors.Wrapf(err, "failed to listen: %v", channelName)
 	}
+	// let the publisher set know the publisher is ready
+	publisher.ReadyChan <- true // XXX (corban): should we close the channel instead?
```

Comment:
hmmm I don't think it's necessary to close all channels, so my opinion is that sending `true` is more clear than closing it

Reply:
My issue here is that the channel is only used to communicate one thing once per publisher, so it feels weird to leave it open since it will never be GC'ed (assuming the publisher lives forever); however, if we instead have it as a parameter that gets passed into `p.start()` during `ps.Start()`, then we avoid that issue entirely. 

So, yes, I think sending true is a more intuitive behavior, especially since these aren't buffered channels and a better solution for enforcing how we think about this channel is to make it a parameter rather than a field.

Reply:
A down side to having `ReadChan <- true` + not using a buffered channel, means that we are also blocking the publisher start from moving forward until the publisher set reads that ready status off the channel, which isn't exactly necessary; we don't have that issue if we close the channel.